packages/core/src/constants.ts

```diff
@@ -54,6 +55,12 @@ export const VISUAL_EDITOR_EVENTS = {
   Ready: 'cfVisualEditorReady',
 };
 
+export enum StudioExperienceMode {
```

Comment:
It's kinda funny that the `mode` is having a [comeback](https://github.com/contentful/experience-builder/pull/341/files#diff-c4805c4c6737dc23dbf80cdc5ee3b59903b10e7a76d2859615442821deb714ddR34) in this new shape. We used to have a mode to separate preview, delivery, and editor mode which we deprecated and removed after a while.
Maybe, we should make it somehow clear (e.g. through a comment) that this mode is only for the editor experience inside the iframe and not about preview/ delivery.

Reply:
Good call out, I was not aware of the history of "modes" in studio.  That same question entered my mind yesterday, for what it's worth, I created a poll to see how people referred to "editor/read-only modes".

<img width="706" alt="Screenshot 2024-09-04 at 9 06 14 AM" src="https://github.com/user-attachments/assets/f6cb5002-9dc4-4d19-a833-b5c40e777ea6">

@jsdalton, had a particularly good point, which is what led me decide on `StudioExperienceMode`.

> the canvas view is what you showed, where you can drag and drop and use the toolbars, but in a way it's a subset of the "experience editor" -- which doesn't currently support many other modes of operation other than canvas mode, but could in the future.e

* takes a breather * 😂

All that being said, I agree with you that maybe we more intentionally name this to be just for editor/read-only mode...  In a way that's already clear since the actual values of the enum express it, but that doesn't mean someone couldn't come along 6 months from now and wonder "why doesn't this include preview/delivery?!?!?".

In my weakly held opinion, I like the name `StudioExperienceMode`, or maybe `StudioCanvasExperienceMode`, and a comment should make it exactly clear (hopefully).

Reply:
Thanks for checking and sharing that poll. Very interesting & surprising insight :)
As long as we're aware of this potential confusion, I'm happy with going forward as is.